src/txmempool.cpp

```diff
@@ -48,6 +53,246 @@ CTxMemPoolEntry::GetPriority(unsigned int currentHeight) const
     return dResult;
 }
 
+// Update the given tx for any in-mempool descendants.
+// Assumes that setMemPoolChildren is correct for the given tx and all
+// descendants.
+bool CTxMemPool::UpdateForDescendants(txiter updateIt, int maxDescendantsToVisit, cacheMap &cachedDescendants, const std::set<uint256> &setExclude)
+{
+    // Track the number of entries (outside setExclude) that we'd need to visit
+    // (will bail out if it exceeds maxDescendantsToVisit)
+    int nChildrenToVisit = 0; 
+
+    setEntries stageEntries, setAllDescendants;
+    stageEntries = GetMemPoolChildren(updateIt);
+
+    while (!stageEntries.empty()) {
+        setAllDescendants.insert(stageEntries.begin(), stageEntries.end());
+
+        setEntries entriesToAdd;
+        BOOST_FOREACH(const txiter cit, stageEntries) {
+            if (cit->IsDirty()) {
```

Comment:
Dirty implies something changed typically,  perhaps instead change this to `ShouldSkip`?